### DISCUSSION TITLE
Invalid file path

### DISCUSSION THREAD

**Discussion by ellenalamont17** [earthlab-education/Earth-Analytics-2023-01-Intro#686] [! GitHub help] [answered]
I am trying to open my srtm data using a filepath to the downloaded data saved in a data dictionary.  The dictionary has key = grassland_name and values = srtm_filepaths. The output of printing the srtm_paths dictionary is shown below. 

![image](https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/143370562/2f254425-5ac7-4478-b94a-9be419266c0a)

While troubleshooting my function, I tried to create a simple loop that would open the file paths for each of the grasslands and print the resulting array for each unit. 
``````
for g_name, path in srtm_paths.items():
    srtm_da = rxr.open_rasterio(path, masked=True).squeeze()
    print(srtm_da)
``````
However, just like with my function, I get an error stating that the path is invalid.
![image](https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/143370562/baf86a7b-5d06-4f24-bb8a-ca9383053b7d)

I have not idea what is wrong here.

The filepath as shown in my windows file manager is shown below:

C:\Users\elamont\earth-analytics\data\srtm-elevation\butte-valley-national-grassland\butte-valley-national-grassland\SRTMGL1_NC.003_2000001_to_2023351\SRTMGL1_NC.003_SRTMGL1_DEM_doy2000042_aid0001.tif

**Comment by edchanonline** [comment, marked as answer]
Make sure your paths are not embedded in a list.

To illustrate - 

Snippet: 
```
grasslands[0]['srtm_paths'], grasslands[1]['srtm_paths']
```
Result:
```
(['/Users/ed.chan/earth-analytics/data/sorghastrum-nutans/srtm-oglala/SRTMGL1_NC.003_2000001_to_2023345/SRTMGL1_NC.003_SRTMGL1_DEM_doy2000042_aid0001.tif'],
 ['/Users/ed.chan/earth-analytics/data/sorghastrum-nutans/srtm-pawnee/SRTMGL1_NC.003_2000001_to_2023345/SRTMGL1_NC.003_SRTMGL1_DEM_doy2000042_aid0001.tif'])
```

Snippet: 
```
grasslands[0]['srtm_paths'][0], grasslands[1]['srtm_paths'][0]
```
Result:
```
('/Users/ed.chan/earth-analytics/data/sorghastrum-nutans/srtm-oglala/SRTMGL1_NC.003_2000001_to_2023345/SRTMGL1_NC.003_SRTMGL1_DEM_doy2000042_aid0001.tif',
 '/Users/ed.chan/earth-analytics/data/sorghastrum-nutans/srtm-pawnee/SRTMGL1_NC.003_2000001_to_2023345/SRTMGL1_NC.003_SRTMGL1_DEM_doy2000042_aid0001.tif')
```

My actual line at opening these files:
```
elevation_data = [rxr.open_rasterio(path, masked=True).squeeze() for path in grasslands[index]['srtm_paths']][0]
```
